main.go

```diff
@@ -235,24 +258,52 @@ func main() {
 			})
 		}
 
-		if cfg.url != nil {
+		if cfg.upstream.url != nil {
 			ctx, cancel := context.WithCancel(ctx)
 			// Create Reverse Proxy.
 			p := httputil.ReverseProxy{
 				Director: func(request *http.Request) {
-					request.URL.Scheme = cfg.url.Scheme
+					request.URL.Scheme = cfg.upstream.url.Scheme
 					// Set the Host at both request and request.URL objects.
-					request.Host = cfg.url.Host
-					request.URL.Host = cfg.url.Host
+					request.Host = cfg.upstream.url.Host
+					request.URL.Host = cfg.upstream.url.Host
 					// Derive path from the paths of configured URL and request URL.
-					request.URL.Path, request.URL.RawPath = joinURLPath(cfg.url, request.URL)
+					request.URL.Path, request.URL.RawPath = joinURLPath(cfg.upstream.url, request.URL)
 					// Add prefix header with value "/", since from a client's perspective
-					// we are forwarding /<anything> to /<cfg.url.Path>/<anything>.
+					// we are forwarding /<anything> to /<cfg.upstream.url.Path>/<anything>.
 					request.Header.Add(prefixHeader, "/")
 				},
 			}
+
+			base := http.DefaultTransport
+			if cfg.upstream.caFile != "" {
+				caCert, err := ioutil.ReadFile(cfg.upstream.caFile)
+				if err != nil {
+					stdlog.Fatalf("failed to initialize upstream server TLS CA: %v", err)
+				}
+				pool := x509.NewCertPool()
+				pool.AppendCertsFromPEM(caCert)
+
+				base = &http.Transport{
+					Proxy: http.ProxyFromEnvironment,
```

Comment:
Can we use the default transport from the stdlib and clone it and add the TLS config rather than hardcore in all of these values?